### PR TITLE
feat: [BB-4615] Add custom tags to calculate the tax related values

### PR DESCRIPTION
**Pull request by alfredchavez**
This PR adds a custom template tag that allows the calculation of taxes applied to a product from its price, by multiplying this value by the TAX_RATE defined in settings

Jira Issue: https://tasks.opencraft.com/browse/BB-4615

### CODE REVIEW

**Comment by lgp171188** [comment]
@alfredchavez, is this PR ready for review? Asking because the title still has `WIP`.

File: ecommerce/extensions/checkout/views.py
Comment (lgp171188) [inline ecommerce/extensions/checkout/views.py]: @alfredchavez, can you explain why this is needed?

File: ecommerce/extensions/basket/templatetags/basket_tags.py
Comment (lgp171188) [inline ecommerce/extensions/basket/templatetags/basket_tags.py]: @alfredchavez, shouldn't we ensure that there are just 2 digits after the decimal place, irrespective of how many digits are there in the tax rate after the decimal point?

Comment (alfredchavez) [reply ecommerce/extensions/basket/templatetags/basket_tags.py]: Sure, I thought as currency does it by its own, it would not be needed, but better to be sure. I've just updated it so it rounds it up to 2 decimal places

File: ecommerce/extensions/checkout/views.py
Comment (alfredchavez) [reply ecommerce/extensions/checkout/views.py]: @lgp171188 I've added it so user full name can be used on the receipt page. As I tried previously with user.full_name and order.user.full_name and I could not get to display the value

File: ecommerce/extensions/checkout/views.py
Comment (lgp171188) [inline ecommerce/extensions/checkout/views.py]: @alfredchavez, (CC @giovannicimolin), I don't think it is a good idea to add this to our common branch since it is specific to one of our customers.

We could try adding a template tag in the pluggable PayTabs app that retrieves the Arabic name from LMS for the given user. This will require that app to be in the `INSTALLED_APPS` setting and we have `ADDL_INSTALLED_APPS` that can be overridden using Ansible configuration.

Comment (alfredchavez) [reply ecommerce/extensions/checkout/views.py]: @lgp171188  I've managed to add a tag that gets the arabic name and displays it in a custom tag, but I had to update the packages versions in the requirements(django and ipware, as koa.3 ecommerce uses django 2.20) of https://github.com/open-craft/ecommerce-paytabs, not sure how that will impact other clients, let me know if that is ok, I am going to open a PR in that repo
Or perhaps is it better to add a custom template tag directly to ecommerce?

Comment (alfredchavez) [reply ecommerce/extensions/checkout/views.py]: @lgp171188 and I cannot push a new branch to that repo, I need permissions there 
`ERROR: Permission to open-craft/ecommerce-paytabs.git denied to alfredchavez`

Comment (giovannicimolin) [reply ecommerce/extensions/checkout/views.py]: @alfredchavez The correct repo is on GitLab: https://gitlab.com/opencraft/client/kkux/ecommerce-paytabs.
Use that one instead.

Comment (giovannicimolin) [reply ecommerce/extensions/checkout/views.py]: @alfredchavez I've given you access to it.

Comment (alfredchavez) [reply ecommerce/extensions/checkout/views.py]: Thanks @giovannicimolin :)

File: ecommerce/extensions/checkout/views.py
Comment (lgp171188) [inline ecommerce/extensions/checkout/views.py]: @alfredchavez, CC @giovannicimolin, Arabic name is not an extended profile field since it comes from the ExtraInfo model, right? So this wouldn't work?

File: ecommerce/extensions/basket/templatetags/basket_tags.py
Comment (lgp171188) [inline ecommerce/extensions/basket/templatetags/basket_tags.py:26]: @alfredchavez, nit: missing docstring

File: ecommerce/extensions/basket/templatetags/basket_tags.py
Comment (lgp171188) [inline ecommerce/extensions/basket/templatetags/basket_tags.py:35]: @alfredchavez, nit: missing docstring

File: ecommerce/extensions/basket/templatetags/basket_tags.py
Comment (lgp171188) [inline ecommerce/extensions/basket/templatetags/basket_tags.py]: @alfredchavez, nit: missing module docstring.

File: ecommerce/extensions/basket/templatetags/basket_tags.py
Comment (lgp171188) [inline ecommerce/extensions/basket/templatetags/basket_tags.py:43]: @alfredchavez, nit: missing docstring

File: ecommerce/extensions/checkout/views.py
Comment (lgp171188) [inline ecommerce/extensions/checkout/views.py]: @alfredchavez, use the multi-line docstring format recommended in our coding standards.

```suggestion
        """
        Helper function to extract data from the extended profile.
        """
```

File: ecommerce/extensions/checkout/views.py
Comment (giovannicimolin) [reply ecommerce/extensions/checkout/views.py]: @lgp171188 It's an extended profile field now. `ExtraInfo` app migrates the data to the user profile after the registration is complete.

https://gitlab.com/opencraft/client/kkux/custom-registration-fields/-/blob/master/custom_reg_form/signals.py#L13

Comment (lgp171188) [reply ecommerce/extensions/checkout/views.py]: @giovannicimolin, neat! 👍🏼

File: ecommerce/extensions/basket/templatetags/basket_tags.py
Comment (alfredchavez) [reply ecommerce/extensions/basket/templatetags/basket_tags.py:43]: @lgp171188 thanks, I've just added the missing docstrings :)

File: ecommerce/extensions/checkout/views.py
Comment (lgp171188) [reply ecommerce/extensions/checkout/views.py]: @alfredchavez, @giovannicimolin, this will return `None` for the `arabic_name` extended profile field unless it is configured in the `extended_profile_field` `SiteConfiguration` parameter. I don't see this set up on the production instance.

**Review by lgp171188 — APPROVED** [review]
@alfredchavez, this can be merged after the open conversation about the extended profile fields is resolved :+1: You will still have to move the custom code to the PayTabs plugin app in this sprint after this PR is merged. FYI.
- [x] I tested the changes in this PR and verified that they work okay and provide all the details to the receipt and bucket pages.
- [x] I read through the code
- [x] ~~I checked for accessibility issues~~ NA
- [x] ~~Includes documentation~~ NA

File: ecommerce/extensions/checkout/views.py
Comment (alfredchavez) [reply ecommerce/extensions/checkout/views.py]: @lgp171188 would it be better to return an empty string(instead of the `None`)?  CC @giovannicimolin

File: ecommerce/extensions/checkout/views.py
Comment (lgp171188) [inline ecommerce/extensions/checkout/views.py]: @alfredchavez, I am wondering if it makes sense to return the full name instead? CC @giovannicimolin

Comment (alfredchavez) [reply ecommerce/extensions/checkout/views.py]: @lgp171188  mmm, yes, I was also considering it, I am going to change it to return the full_name in case arabic_name is not defined
CC @giovannicimolin

Comment (giovannicimolin) [reply ecommerce/extensions/checkout/views.py]: @alfredchavez Yup, full name makes sense when there's no Arabic name available.

Comment (alfredchavez) [reply ecommerce/extensions/checkout/views.py]: @lgp171188 I've changed it so it uses the order user full name instead of arabic name in case it is not available, 
also fixed a "bug" that I just noticed, I've been using the current request user for the invoice instead of the order user, which I have now fixed, let me know if more change is needed here 🙂

Comment (alfredchavez) [reply ecommerce/extensions/checkout/views.py]: @lgp171188 I've changed it so it uses the order user full name instead of arabic name in case it is not available, 
also fixed a "bug" that I just noticed, I've been using the current request user for the invoice instead of the order user, which I have now fixed, let me know if more change is needed here 🙂 CC @giovannicimolin

File: ecommerce/extensions/checkout/views.py
Comment (lgp171188) [inline ecommerce/extensions/checkout/views.py]: @alfredchavez, can you remove the code already moved to the `ecommerce-paytabs` pluggable app before we can merge this?

Comment (alfredchavez) [reply ecommerce/extensions/checkout/views.py]: ok

**Pull request merged**
